client/modules/IDE/components/Editor/hinter.js

```diff
@@ -0,0 +1,118 @@
+import Fuse from 'fuse.js';
+import CodeMirror from 'codemirror';
+import { JSHINT } from 'jshint';
+import { HTMLHint } from 'htmlhint';
+import { CSSLint } from 'csslint';
+
+import 'codemirror/addon/hint/css-hint';
+import * as hinterDefinition from '../../../../utils/p5-hinter';
+import '../show-hint'; // TODO: Remove for codemirror v6?
+
+window.JSHINT = JSHINT;
+window.CSSLint = CSSLint;
+window.HTMLHint = HTMLHint;
+
+const hinter = new Fuse(hinterDefinition.p5Hinter, {
+  threshold: 0.05,
+  keys: ['text']
+});
+
+/** Hides the hinter. */
+export function hideHinter() {
+  CodeMirror.showHint(this._cm, () => {}, {});
```

Comment:
I think this line is causing errors when the Autocomplete Hinter is toggled "off" in the Preferences, since the cm instance is undefined!

Reply:
omg!!! my apologies, great catch 🙏